apps/system/test/unit/statusbar_test.js

```diff
@@ -658,6 +484,12 @@ suite('system/Statusbar', function() {
       assert.isTrue(Statusbar.element.classList.contains('fullscreen-layout'));
       assert.isTrue(stub.calledOnce);
     });
+
+    test('setAppearance maximized with fewIcons', function() {
+      Statusbar._numMinimizedIcons = 5;
+      Statusbar.setAppearance();
+      assert.isFalse(Statusbar.element.classList.contains('maximized'));
+    });
```

Comment:
we also need a test showing that when dispatching an `iconshow` event the `maximized` class will be set.